File: DependencyInjection/ServiceLocatorDIAntipattern/_2_ServiceLocator/Inbound/UdpSocket.cs
using System;

namespace ServiceLocatorDIAntipattern._2_ServiceLocator.Inbound
{
  internal interface IInputSocket
  {
    bool Receive(out byte[] frameData);
  }

  class UdpSocket : IInputSocket
  {
    public bool Receive(out byte[] frameData)
    {
      frameData = new byte[100];
      new Random().NextBytes(frameData);
      return true;
    }
  }
}

File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/Inbound/UdpSocket.java
package com.github.grzesiek_galezowski.BastardInjection.Inbound;

class UdpSocket implements IInputSocket {
  public boolean receive(byte[] frameData) {
    //todo generate random the java way new Random().NextBytes(frameData);
    return true;
  }

  public void close() {
    System.out.println("Disposed");
  }
}
